activesupport/lib/active_support/core_ext/time/calculations.rb

```diff
@@ -217,7 +217,11 @@ def ago(seconds)
   # Returns a new Time representing the time a number of seconds since the instance time
   def since(seconds)
     self + seconds
-  rescue
+  rescue TypeError
+    ActiveSupport.deprecator.warn(
+      "Passing an instance of #{seconds.class} to #{self.class}#since is deprecated. This behavior will raise " \
+      "a `TypeError` in Rails 8.0."
```

Comment:
As `main` is 8.0 already this would get deprecated in 8.1:
```suggestion
      "a `TypeError` in Rails 8.1."
```

Reply:
Should I update this too https://github.com/rails/rails/pull/52343/files#diff-aa0ae5ccf92f812f874b632afe70375c52772636f927fe6e34ffeaebf54af9d1R308

Reply:
I think so, but there are core members reviewing in that PR so now I'm not 100% sure 😁 
Maybe it can be added to the 7.2 release but I'm not sure how important this change is.